app/src/main/java/com/infomaniak/mail/data/models/thread/Thread.kt

```diff
@@ -86,19 +91,31 @@ class Thread : RealmObject {
 
         // TODO: When do we want to update this value? This is a quick fix. The date will
         // TODO: only update when we get data from the API. We probably don't want that.
-        displayedDate = formatDate(date?.toDate() ?: Date(0))
+        displayedDate = date?.toDate()?.formatDate() ?: ""
 
         return this
     }
 
-    private fun formatDate(date: Date): String = with(date) {
-        when {
+    private fun Date.formatDate(): String {
+        return when {
             isToday() -> format(FORMAT_DATE_HOUR_MINUTE)
             year() == Date().year() -> format(FORMAT_DATE_SHORT_DAY_ONE_CHAR)
             else -> format(FORMAT_DATE_CLEAR_MONTH_DAY_ONE_CHAR)
         }
     }
 
+    fun formatExpeditorField(context: Context): CharSequence {
+        return buildSpannedString {
+            if (hasDrafts) {
+                color(context.getColor(R.color.draftTextColor)) {
+                    append("(${context.getString(R.string.messageIsDraftOption)}) ")
+                }
+            }
+            val recipients = if (hasDrafts) to else from
+            recipients.forEach { append("${it.displayedName(context)}, ") }
+        }.removeSuffix(", ")
```

Comment:
you can replace this with
```suggestion
            append(recipients.joinToString { it.displayedName(context)  })
```

Reply:
done